presto-memory-context/src/main/java/com/facebook/presto/memory/context/SimpleLocalMemoryContext.java

```diff
@@ -76,6 +76,16 @@ public synchronized boolean trySetBytes(long bytes)
         return false;
     }
 
+    /**
+     * This method transfers the allocations from this memory context to the "to" memory context,
+     * where parent of this is a descendant of to.parent (there can be multiple AggregatedMemoryContexts between them).
+     * <p>
+     * During the transfer the implementation of this method must not reflect any state changes outside of the contexts
+     * (e.g., by calling the reservation handlers).
+     * <p>
+     * This method currently has a single use ({@code NestedLoopJoinPages}) and any change should be tested carefully due to
+     * its somewhat complex semantics.
```

Comment:
I'd love to see this `@Deprecated`. It's too brittle.

Reply:
OK I am marking it as `Deprecated`.